doc/ansible/dev/developer_guide.md

```diff
@@ -0,0 +1,430 @@
+
+**playbook**:  This is the path to the playbook that you have added to the
+container. This playbook is expected to be simply a way to call roles. This
+field is mutually exclusive with the "role" field.
+
+**role**:  This is the path to the role that you have added to the container.
+For example if your roles directory is at `/opt/ansible/roles/` and your role
+is named `busybox`, this value will be `/opt/ansible/roles/busybox`. This field
+is mutually exclusive with the "playbook" field.
+
+Example specifying a role:
+
+```yaml
+---
+- version: v1alpha1
+  group: foo.example.com
+  kind: Foo
+  role: /opt/ansible/roles/Foo
+```
+
+#### Using playbooks in watches.yaml
+
+By default, `operator-sdk new --type ansible` sets `watches.yaml` to execute a
+role directly on a resource event. This works well for new projects, but with a
+lot of Ansible code this can be hard to scale if we are putting everything
+inside of one role. Using a playbook allows the developer to have more
+flexibility in consuming other roles and enabling more customized deployments
+of their application. To do this, modify `watches.yaml` to use a playbook
+instead of the role:
+```yaml
+---
+- version: v1alpha1
+  group: foo.example.com
+  kind: Foo
+  playbook: /opt/ansible/playbook.yml
+```
+
+Modify `tmp/build/Dockerfile` to put `playbook.yml` in `/opt/ansible` in the
+container in addition to the role (`/opt/ansible` is the `HOME` environment
+variable inside of the Ansible Operator base image):
+```Dockerfile
+FROM quay.io/water-hole/ansible-operator
+
+COPY roles/ ${HOME}/roles
+COPY playbook.yaml ${HOME}/playbook.yaml
+COPY watches.yaml ${HOME}/watches.yaml
+```
+
+Alternatively, to generate a skeleton project with the above changes, a
+developer can also do:
+```bash
+$ operator-sdk new --type ansible --kind Foo --api-version foo.example.com/v1alpha1 foo-operator --generate-playbook
+```
+
+### Custom Resource file
+
+The Custom Resource file format is Kubernetes resource file. The object has
+mandatory fields:
+
+**apiVersion**:  The version of the Custom Resource that will be created.
+
+**kind**:  The kind of the Custom Resource that will be created
+
+**metadata**:  Kubernetes specific metadata to be created
+
+**spec**:  This is the key-value list of variables which are passed to Ansible.
+This field is optional and will be empty by default.
+
+**annotations**: Kubernetes specific annotations to be appened to the CR. See
+the below section for Ansible Operator specifc annotations.
+
+#### Ansible Operator annotations
+This is the list of CR annotations which will modify the behavior of the operator:
+
+**ansible.operator-sdk/reconcile-period**: Used to specify the reconciliation
+interval for the CR. This value is parsed using the standard Golang package
+[time][time_pkg]. Specifically [ParseDuration][time_parse_duration] is used
+which will use the default of `s` suffix giving the value in seconds.
```

Comment:
s/default of s suffix/default suffix of s/
s/use/apply/ just for writing aesthetics